test/zimcheck-test.cpp

```diff
@@ -66,6 +66,18 @@ TEST(zimfilechecks, test_articles)
     ASSERT_TRUE(logger.overallStatus());
 }
 
+TEST(zimfilechecks, test_redirect_loop)
+{
+  std::string fn = "data/zimfiles/wikibooks_be_all_nopic_2017-02.zim";
+
+  zim::Archive archive(fn);
+  ErrorLogger logger;
+
+  test_redirect_loop(archive, logger);
+
+  ASSERT_TRUE(logger.overallStatus());
+}
```

Comment:
This is half of testing the new code. It only proves that if there are no redirect loops, the check passes. An empty function body  behind `test_redirect_loop()` would provide the same outcome. You must also verify that if there *are* redirect loops, the check fails.